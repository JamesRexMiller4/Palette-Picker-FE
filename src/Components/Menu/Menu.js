import React, { useEffect } from 'react';
import { useStore } from '../App/App.js';
import './Menu.scss';
import { Link } from 'react-router-dom';
import shuffle from '../../Images/001-random.svg';
import heart from '../../Images/002-heart.svg'
import folder from '../../Images/003-folder.svg'

export const Menu = () => {
  const { state, dispatch } = useStore();

  useEffect(() => {
    
  })

  return (
    <nav className='vertical-menu menu'>
      <Link to={'/folders'}>
        <img className='icon' src={folder} alt='icon of a folder'></img>
      </Link>
      <input type='text' placeholder='Create New Folder'></input>
      <p>or</p>
      <select>
        <option>project one</option>
      </select>
      <input type='text' placeholder='Name This Palette'></input>
      <img className='icon' src={heart} alt='icon of a heart'></img>
      <img  className='icon' src={shuffle} alt='shuffle icon' onClick={() => dispatch({type: 'GENERATE COLORS', payload: state})}></img>
    </nav>
  )
}

export default Menu;