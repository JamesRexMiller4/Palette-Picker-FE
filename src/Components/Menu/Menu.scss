.vertical-menu {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  width: 15vw;
  height: 79.75vh;
  border: 1px solid lavender;
  padding: 5vh 0;
}

.horizontal-menu {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  width: 15vw;
  height: 79.75vh;
  border: 1px solid lavender;
  padding: 5vh 0;
}

.icon {
  width: 5vw;
    &:hover {
      cursor: pointer;
    }
}

