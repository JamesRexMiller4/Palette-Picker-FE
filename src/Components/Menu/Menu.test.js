import React from 'react';
import Menu from './Menu';
import { shallow } from 'enzyme';

describe('App', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<Menu />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})