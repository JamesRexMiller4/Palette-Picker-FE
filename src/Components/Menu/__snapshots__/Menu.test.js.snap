// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<nav
  className="vertical-menu menu"
>
  <Link
    to="/folders"
  >
    <img
      alt="icon of a folder"
      className="icon"
      src="003-folder.svg"
    />
  </Link>
  <input
    name="create_folder_input"
    onChange={[Function]}
    placeholder="Create New Folder"
    type="text"
  />
  <button
    name="create_folder"
    onClick={[Function]}
    type="button"
  >
    Create New Folder
  </button>
  <p>
    Save Palette
  </p>
  <select
    name="choose_folder"
    onChange={[Function]}
  >
    <option>
      choose existing folder
    </option>
  </select>
  <input
    name="name_palette"
    onChange={[Function]}
    placeholder="Name This Palette"
    type="text"
    value=""
  />
  <img
    alt="icon of a heart"
    className="icon"
    name="heart"
    onClick={[Function]}
    src="002-heart.svg"
  />
  <img
    alt="shuffle icon"
    className="icon"
    name="shuffle"
    onClick={[Function]}
    src="001-random.svg"
  />
</nav>
`;
